core/markets/storage/mk_protocol.hpp

```diff
@@ -0,0 +1,399 @@
+    std::string transfer_type;
+    CID root;
+    // Optional for non-manual transfer, will be recomputed from the data if not
+    // given
+    boost::optional<CID> piece_cid;
+    // Optional for non-manual transfer, will be recomputed from the data if not
+    // given
+    UnpaddedPieceSize piece_size;
+    // Optional: used as the denominator when calculating transfer
+    uint64_t raw_block_size{};
+  };
+
+  /** For protocol v1.0.1. */
+  struct DataRefV1_0_1 : public DataRef {};
+
+  CBOR_TUPLE(DataRef, transfer_type, root, piece_cid, piece_size)
+
+  /** Protocol V1.1.0 with named encoding and extended DataRef */
+  struct DataRefV1_1_0 : public DataRef {};
+
+  inline CBOR2_ENCODE(DataRefV1_1_0) {
+    auto m{CborEncodeStream::map()};
+    m["TransferType"] << v.transfer_type;
+    m["Root"] << v.root;
+    m["PieceCid"] << v.piece_cid;
+    m["PieceSize"] << v.piece_size;
+    m["RawBlockSize"] << v.raw_block_size;
+    return s << m;
+  }
+
+  inline CBOR2_DECODE(DataRefV1_1_0) {
+    auto m{s.map()};
+    CborDecodeStream::named(m, "TransferType") >> v.transfer_type;
+    CborDecodeStream::named(m, "Root") >> v.root;
+    CborDecodeStream::named(m, "PieceCid") >> v.piece_cid;
+    CborDecodeStream::named(m, "PieceSize") >> v.piece_size;
+    CborDecodeStream::named(m, "RawBlockSize") >> v.raw_block_size;
+    return s;
+  }
+
+  enum class StorageDealStatus : uint64_t {
+    STORAGE_DEAL_UNKNOWN = 0,
+    STORAGE_DEAL_PROPOSAL_NOT_FOUND,
+    STORAGE_DEAL_PROPOSAL_REJECTED,
+    STORAGE_DEAL_PROPOSAL_ACCEPTED,
+    STORAGE_DEAL_STAGED,
+    STORAGE_DEAL_SEALING,
+    STORAGE_DEAL_FINALIZING,
+    STORAGE_DEAL_ACTIVE,
+    STORAGE_DEAL_EXPIRED,
+    STORAGE_DEAL_SLASHED,
+    STORAGE_DEAL_REJECTING,
+    STORAGE_DEAL_FAILING,
+    // Internal
+
+    /** Deposited funds as necessary to create a deal, ready to move forward */
+    STORAGE_DEAL_FUNDS_ENSURED,
+    STORAGE_DEAL_CHECK_FOR_ACCEPTANCE,
+    /** Verifying that deal parameters are good */
+    STORAGE_DEAL_VALIDATING,
+    STORAGE_DEAL_ACCEPT_WAIT,
+    STORAGE_DEAL_START_DATA_TRANSFER,
+    /** Moving data */
+    STORAGE_DEAL_TRANSFERRING,
+
+    /** Manual transfer */
+    STORAGE_DEAL_WAITING_FOR_DATA,
+
+    /** Verify transferred data - generate CAR / piece data */
+    STORAGE_DEAL_VERIFY_DATA,
+
+    /** Ensuring that provider collateral is sufficient */
+    STORAGE_DEAL_ENSURE_PROVIDER_FUNDS,
+
+    /** Ensuring that client funds are sufficient */
+    STORAGE_DEAL_ENSURE_CLIENT_FUNDS,
+
+    /** Waiting for funds to appear in Provider balance */
+    STORAGE_DEAL_PROVIDER_FUNDING,
+
+    /** Waiting for funds to appear in Client balance */
+    STORAGE_DEAL_CLIENT_FUNDING,
+
+    /** Publishing deal to chain */
+    STORAGE_DEAL_PUBLISH,
+
+    /** Waiting for deal to appear on chain */
+    STORAGE_DEAL_PUBLISHING,
+
+    /** deal failed with an unexpected error */
+    STORAGE_DEAL_ERROR,
+  };
+
+  /** Base MinerDeal */
+  struct MinerDeal {
```

Comment:
there are more fields

https://github.com/filecoin-project/go-fil-markets/blob/v1.13.1/storagemarket/migrations/migrations.go#L49-L69

https://github.com/filecoin-project/go-fil-markets/blob/v1.13.1/storagemarket/types.go#L91-L115

Reply:
MinerDeal and ClientDeal are internal provider states of the deals and not a part of the protocol. I'm going to move it away from protocol headers.